synapseclient/models/protocols/team_protocol.py

```diff
@@ -0,0 +1,139 @@
+"""Protocol for the specific methods of this class that have synchronous counterparts
+generated at runtime."""
+
+from typing import Optional, TYPE_CHECKING, Protocol, List, Dict
+from synapseclient import Synapse
+
+if TYPE_CHECKING:
+    from synapseclient.models import TeamMember, Team
+
+
+class TeamSynchronousProtocol(Protocol):
+    """
+    The protocol for methods that are asynchronous but also
+    have a synchronous counterpart that may also be called.
+    """
+
+    def create(self, synapse_client: Optional[Synapse] = None) -> "Team":
+        """Creates a new team on Synapse.
+
+        Arguments:
+            synapse_client: If not passed in or None this will use the last client
+                from the `.login()` method.
+
+        Returns:
+            Team: The Team object.
+        """
+        return self
+
+    def delete(self, synapse_client: Optional[Synapse] = None) -> None:
+        """Deletes a team from Synapse.
+
+        Arguments:
+            synapse_client: If not passed in or None this will use the last client
+                from the `.login()` method.
+
+        Returns:
+            None
+        """
+        return None
+
+    def get(self, synapse_client: Optional[Synapse] = None) -> "Team":
+        """Gets a Team from Synapse.
```

Comment:
Need to specify how a Team is retrieved with this call (id and/or name) - I know it's implied under `Raises`, but just to make it a little more clear. Also, what happens with `id` and `name` are fed but one does not match the other?